hibernate-core/src/main/java/org/hibernate/procedure/internal/ParameterBindImpl.java

```diff
@@ -78,10 +78,12 @@ private void internalSetValue(T value) {
 			throw new IllegalStateException( "Can only bind values for IN/INOUT parameters : " + procedureParameter );
 		}
 
-		if ( procedureParameter.getParameterType() != null ) {
-			if ( !procedureParameter.getParameterType().isInstance( value ) ) {
-				throw new IllegalArgumentException( "Bind value [" + value + "] was not of specified type [" + procedureParameter.getParameterType() );
-			}
+		if ( value == null && !procedureParameter.isPassNullsEnabled() ) {
+			throw new IllegalArgumentException( "Parameter '" + procedureParameter.getName() +  "' doesn't allow NULL value" );
+		}
+		
+		if ( value != null && !procedureParameter.getParameterType().isInstance( value ) ) {
```

Comment:
Is there any particular reason why your change dropped the original outer if-check here?

Reply:
Yes, because It is unnecessary due to that the parameter type at this code point never be null.
If you register the parameter using annotation `@StoredProcedureParameter` it is mandatory, and if you try to create it programmatically avoiding to set the type, the app will throw a `java.lang.NullPointerException` because of code `type.getName()` at line 322 of class `org.hibernate.procedure.internal.ProcedureCallImpl`

Reply:
Thanks, could you please attach a test case that reproduces the NPE with your fix and open a JIRA if you have not already done so for this change?

Reply:
Ok, I will have a delay to attach the test because I need to recreate the same problem in a in-memory db, building a stored procedure with nullable parameter, and calling the sp using a null value in the parameter.

Reply:
@matsebas Any update?

Reply:
Sorry Chris I’ve been working full-time, I will try to do unit tests next week.

Reply:
@matsebas still interested in a test case. It looks like a good fix but we really need a test case to include it.

Thanks!

Reply:
Hi Guillaume (@gsmet), I return to code time, and I look that this PR should be closed because it was resolved in commit https://github.com/hibernate/hibernate-orm/commit/cc1ffbaaa066df990d34aea6cb4ae59f3bbfbaf4

Reply:
@matsebas OK, perfect, thanks for the follow-up!